### ISSUE TITLE
Going back in the Inspector while editing a GridMap node makes it and the previous node selected

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot#21297]
**Godot version:**
 2db4942

**Issue description:**
Basically this:
![screenshot at 2018-08-22 21-46-22](https://user-images.githubusercontent.com/30739239/44498458-f1438d80-a654-11e8-9a3b-25a5a45b2ef7.png)

**Comment by vazzup** [comment]
Hi, I'd like to help. Could you specify steps to recreate the bug?

**Comment by YeldhamDev** [comment]
@vazzup Basically what's said in the title, select a GridMap node, go back in the inspector, using these:
![screenshot at 2018-08-23 11-27-26](https://user-images.githubusercontent.com/30739239/44531646-b0d12780-a6c7-11e8-8e7c-615e3de64675.png)

And both the GridMap and the previous node will be selected.

**Comment by vazzup** [comment]
Ok, thanks. Will look into it

**Comment by vazzup** [comment]
Yes, it is also a bug in the master branch. Also, if there are multiple previous nodes, all of them get selected.

**Comment by vazzup** [comment]
I'm looking at the editor_node.cpp file and I think the change will be around line no. 1476 in the _edit_current function.

**Comment by vazzup** [comment]
Hi, after digging deep, I'm a bit stuck with this bug. I think the bug may be from the EditorInspector::update_tree function, but I can't say for sure. Can someone else please point me in the right direction?

**Comment by avril-gh** [comment]
> I've also seen that this works with more nodes than just the gridmap node.

theres also strange behavior while debugging remote tree (which is running on android device),
When one select a node (in debugger) , other nodes will select itself randomly, and theres no way to stop it, thus no way to use this feature to debug / change properties in inspector at runtime, ect.
Its not easy to reproduce it on simple project with fiew nodes created in editor,
but it begin to appear in projects with higher count of nodes created at runtime.

Just mentioning it here, since it sounds similar, ...but perhaps i should make a new issue for this.

**Comment by DeuxAlpha** [comment]
Hi! I'm able to reproduce the bug, but it doesn't seem to be related to GridMaps only. In my case, any node will cause this issue. It doesn't seem to influence functionality, however, since the inspector displays the correct Node and its properties. The only issue is that several nodes _look_ highlighted. Additionally, the further you go back in history, the more Nodes get highlighted (once again, not influencing functionality, but aesthetics only).

**Comment by KoBeWi** [comment]
Still valid in 4e0f31a
As mentioned above, this applies not only to GridMap, but also mixed-type nodes, e.g. when you select Node2D then Spatial, then go back and then forward (going back doesn't seem to trigger it), both get selected. Might be somehow related to #36291

**Comment by KoBeWi** [comment]
Seems fixed (tested in 4.5).
